### PR TITLE
MINOR: Tweak implementation of `FetchRequest.shuffle` and upgrade.html improvements

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

File: docs/upgrade.html [hunk 1]
```diff
@@ -66,12 +66,22 @@
     <li> The new Java Consumer now allows users to search offsets by timestamp on partitions. </li>
     <li> When using an Authorizer and a user hasn't got <b>Describe</b> authorization on a topic, the broker will no longer return TOPIC_AUTHORIZATION_FAILED errors  
          but just UNKNOWN_TOPIC_OR_PARTITION errors, to avoid leaking topic names.</li>
+    <li> Fetch responses have a size limit by default (10 MB for consumers and 50 MB for replication). Note that this is not an absolute maximum as explained in the
```
Comment (hachikuji) [inline docs/upgrade.html]: Is that backwards? I thought the default for consumers was 50MB.

Comment (ijuma) [reply docs/upgrade.html]: Yes, it is. :) Fixed.

File: docs/upgrade.html [hunk 2]
```diff
@@ -66,12 +66,22 @@
     <li> The new Java Consumer now allows users to search offsets by timestamp on partitions. </li>
     <li> When using an Authorizer and a user hasn't got <b>Describe</b> authorization on a topic, the broker will no longer return TOPIC_AUTHORIZATION_FAILED errors  
          but just UNKNOWN_TOPIC_OR_PARTITION errors, to avoid leaking topic names.</li>
+    <li> Fetch responses have a size limit by default (50 MB for consumers and 10 MB for replication). Note that this is not an absolute maximum as explained in the
```
Comment (hachikuji) [inline docs/upgrade.html]: Is it worth pointing out that we previously only had per-partition fetch limits?

**Comment by hachikuji** [comment]
LGTM

**Comment by ijuma** [comment]
Tests passed locally, merged to trunk and 0.10.1 branches.

**Pull request closed without merging**